homeassistant/components/livebox/config_flow.py

```diff
@@ -0,0 +1,164 @@
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return LiveboxOptionsFlowHandler(config_entry)
+
+    def __init__(self):
+        """Initialize the Livebox flow."""
+        self.host = None
+        self.port = None
+        self.username = None
+        self.password = None
+        self.data_schema = None
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        # Use OrderedDict to guarantee order of the form shown to the user
+        from collections import OrderedDict
+
+        self.data_schema = OrderedDict()
+        self.data_schema[vol.Optional(CONF_HOST, default=DEFAULT_HOST)] = str
+        self.data_schema[vol.Optional(CONF_PORT, default=DEFAULT_PORT)] = str
+        self.data_schema[vol.Optional(CONF_USERNAME, default=DEFAULT_USERNAME)] = str
+        self.data_schema[vol.Required(CONF_PASSWORD)] = str
+
+        return self.async_show_form(
+            step_id="init", data_schema=vol.Schema(self.data_schema)
+        )
+
+    async def async_step_init(self, user_input=None):
+        """Handle a flow start."""
+
+        if user_input is not None:
+            self.host = user_input[CONF_HOST]
+            self.port = user_input[CONF_PORT]
+            self.username = user_input[CONF_USERNAME]
+            self.password = user_input[CONF_PASSWORD]
+            return await self.async_step_link()
+
+        return self.async_show_form(
+            step_id="init", data_schema=vol.Schema(self.data_schema)
+        )
+
+    async def async_step_link(self, user_input=None):
+        """Step for link router."""
+
+        errors = {}
+
+        from aiosysbus import Sysbus
+
+        try:
+            box = Sysbus()
+            await box.open(
+                host=self.host,
+                port=self.port,
+                username=self.username,
+                password=self.password,
+            )
+            return await self._entry_from_box(box)
+
+        except AuthenticationRequired:
+            errors["base"] = "register_failed"
+
+        except CannotConnect:
+            LOGGER.error("Error connecting to the Livebox at %s", self.host)
+            errors["base"] = "linking"
+
+        except Exception:  # pylint: disable=broad-except
+            LOGGER.exception("Unknown error connecting with Livebox at %s", self.host)
+            errors["base"] = "linking"
+
+        # If there was no user input, do not show the errors.
+        if user_input is None:
+            errors = {}
+
+        return self.async_show_form(step_id="link", errors=errors)
+
+    async def _entry_from_box(self, box):
+        """Return a config entry from an initialized box."""
+        config = await box.system.get_deviceinfo()
+        box_id = config["status"]["SerialNumber"]
+
+        # Remove all other entries of hubs with same ID or host
```

Comment:
Shouldn't we instead check that we don't add a box with an ID that is already configured?